src/main/java/nl/dtls/fairdatapoint/service/metadata/catalog/CatalogMetadataService.java

```diff
@@ -53,6 +59,22 @@ public CatalogMetadataService(@Value("${metadataProperties.catalogSpecs:}") Stri
         this.parentType = R3D.REPOSITORY;
     }
 
+    @Override
+    public CatalogMetadata retrieve(@Nonnull IRI uri) throws MetadataServiceException {
+        CatalogMetadata catalog = super.retrieve(uri);
+        try {
+            List<IRI> themes = catalogMetadataRepository.getDatasetThemesForCatalog(catalog.getUri());
+            Set<IRI> set = new TreeSet<>(Comparator.comparing(IRI::toString));
+            set.addAll(catalog.getThemeTaxonomys());
+            set.addAll(themes);
+            catalog.setThemeTaxonomys(new ArrayList<>(set));
```

Comment:
How is the theme taxonomy serialized at the moment? As a collection of `IRI`s? I think we might need to capture some more information about it (depends on how we type it maybe).

Reply:
`dcat:themeTaxonomy <https://purl.org/example#theme>, <https://www.wikidata.org/wiki/Q27318> .`

Yes, it's a list of URI. What data do you want to add?

Reply:
Depends on how the serialization is done. The taxonomy instance itself needs some more info (like `skos:ConceptScheme type, and maybe an `rdfs:label`).

Reply:
Should we do it in this PR or should we solve it separately?

Reply:
Probably separately, if it is serialized as it is now we have some ground for discussion.

Reply:
ok